File: scala/src/main/scala/Fibonacci/FibonacciSeries.scala
package fibonacci
import scala.annotation.tailrec

class FibonacciSeries {

  def fibonacciRecursive(position : Int) : List[Int] = position match {
    case 0 => List(1)
    case 1 => List(1, 1)
    case position => {
      val sequence = fibonacciRecursive(position - 1)
      sequence :+ sequence.takeRight(2).sum
    }
  }

  @tailrec final def fibonacciTailRecursive(position : Int, updatedList : List[Int] = List(1) ) : List[Int] = position match {
    case 0 => updatedList
    case position => fibonacciTailRecursive (position - 1, updatedList :+ updatedList.takeRight(2).sum)
  }
}

File: clojure/src/clojure_fibonacci/FibonacciSeries.clj
(ns clojure-fibonacci.FibonacciSeries)

  
(defn fibonacciRecursive [position] 
	(case position
		0 [1]
		1 [1 1]
		(conj 
			@(def numberSequence (fibonacciRecursive(dec position))) (reduce + (take-last 2 numberSequence)))))

			
(defn fibonacciTailRecursive [position] 
	(loop [newPosition position updatedList [1]]
		(if (zero? newPosition) updatedList
			(recur (dec newPosition)  (conj updatedList (reduce + (take-last 2 updatedList)))))))
